sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuHashJoin.scala

```diff
@@ -262,23 +268,76 @@ trait GpuHashJoin extends GpuExec {
     }
   }
 
-  private[this] def doJoinLeftRight(leftTable: Table, rightTable: Table): ColumnarBatch = {
-    val joinedTable = joinType match {
-      case LeftOuter => leftTable.onColumns(joinKeyIndices: _*)
-          .leftJoin(rightTable.onColumns(joinKeyIndices: _*), false)
-      case RightOuter => rightTable.onColumns(joinKeyIndices: _*)
-          .leftJoin(leftTable.onColumns(joinKeyIndices: _*), false)
-      case _: InnerLike => leftTable.onColumns(joinKeyIndices: _*)
-          .innerJoin(rightTable.onColumns(joinKeyIndices: _*), false)
-      case LeftSemi => leftTable.onColumns(joinKeyIndices: _*)
-          .leftSemiJoin(rightTable.onColumns(joinKeyIndices: _*), false)
-      case LeftAnti => leftTable.onColumns(joinKeyIndices: _*)
-          .leftAntiJoin(rightTable.onColumns(joinKeyIndices: _*), false)
-      case FullOuter => leftTable.onColumns(joinKeyIndices: _*)
-          .fullJoin(rightTable.onColumns(joinKeyIndices: _*), false)
-      case _ => throw new NotImplementedError(s"Joint Type ${joinType.getClass} is not currently" +
-          s" supported")
+  private[this] def filterNulls(table: Table, joinKeyIndices: Range, closeTable: Boolean): Table = {
+    var mask: ai.rapids.cudf.ColumnVector = null
+    try {
+      joinKeyIndices.indices.foreach { c =>
+        mask = withResource(table.getColumn(c).isNotNull) { nn =>
+          if (mask == null) {
+            nn.incRefCount()
+          } else {
+            withResource(mask) { _ =>
+              mask.and(nn)
```

Comment:
I think this needs to be a bitwise-OR here instead of bitwise-AND.  If any, not all, of the join key columns are null it shouldn't match.  It will still work with bitwise-AND but it can still lead to long collision chains on null columns and poor performance.

Reply:
Oh nevermind, this is the mask of values to preserve not to filter.